src/runtime/chan.go

```diff
@@ -308,6 +308,9 @@ func (ch *channel) trySend(value unsafe.Pointer) bool {
 		return false
 	}
 
+	critical()
+	defer exitCritical()
```

Comment:
I'm somewhat uncomfortable with `defer` in an interrupt, as there isn't a guarantee `defer` won't allocate. Maybe it's better to be explicit and move the `exitCritical()` to where it should be run.